app/code/Magento/Inventory/Model/SourceCarrierLinkManagement.php

```diff
@@ -75,15 +73,9 @@ public function __construct(
      */
     public function saveCarrierLinksBySource(SourceInterface $source)
     {
-        if (is_array($source->getCarrierLinks())) {
-            try {
-                $this->deleteCurrentCarrierLinks($source);
-                if (!empty($source->getCarrierLinks())) {
-                    $this->saveNewCarrierLinks($source);
-                }
-            } catch (\Exception $e) {
-                throw new StateException(__('Could not update Carrier Links'), $e);
-            }
+        $this->deleteCurrentCarrierLinks($source);
+        if (is_array($source->getCarrierLinks()) && !empty($source->getCarrierLinks())) {
```

Comment:
according to contract method getCarrierLinks return 
`\Magento\InventoryApi\Api\Data\SourceCarrierLinkInterface[]`
why do we check for `is_array` here?